src/core/ext/filters/client_channel/lb_policy/grpclb/client_load_reporting_filter.cc

```diff
@@ -95,22 +98,27 @@ static void start_transport_stream_op_batch(
   GPR_TIMER_SCOPE("clr_start_transport_stream_op_batch", 0);
   // Handle send_initial_metadata.
   if (batch->send_initial_metadata) {
-    // Grab client stats object from user_data for LB token metadata.
-    grpc_linked_mdelem* lb_token =
+    // Grab client stats object from metadata.
+    grpc_linked_mdelem* client_stats_md =
         batch->payload->send_initial_metadata.send_initial_metadata->idx.named
-            .lb_token;
-    if (lb_token != nullptr) {
-      grpc_core::GrpcLbClientStats* client_stats =
-          static_cast<grpc_core::GrpcLbClientStats*>(grpc_mdelem_get_user_data(
-              lb_token->md, grpc_core::GrpcLbClientStats::Destroy));
+            .grpclb_client_stats;
+    if (client_stats_md != nullptr) {
+      const char* client_stats_addr_str = reinterpret_cast<const char*>(
+          GRPC_SLICE_START_PTR(GRPC_MDVALUE(client_stats_md->md)));
+      grpc_core::GrpcLbClientStats* client_stats = nullptr;
+      sscanf(client_stats_addr_str, "%p", &client_stats);
```

Comment:
is it necessary to serialize the pointer into a string?

Reply:
I couldn't think of another way to do it.  I'd welcome alternative suggestions.

Reply:
I have  two random ideas (please feel free to ignore):
1) `StringView(ptr, 0)` is a valid pointer.  We can potentially pass that as the value instead. `grpc_slice_from_static_string` respects that, so we can send the pointer as a string. :-)
2) We can add a `Get/SetStat()` to `MetadataInterface`.

Reply:
Awesome! Thank you!